src/Widgets/PaymentPlan.tsx

```diff
@@ -99,12 +104,16 @@ const PaymentPlanWidget: React.FC<Props> = ({
         >
           {eligibilityPlans.length !== 0 && paymentPlanInfoText(eligibilityPlans[current])}
         </div>
-      </button>
-      <EligibilityModal
-        isOpen={isOpen}
-        onClose={closeModal}
-        eligibilityPlans={eligibilityPlans.filter((plan) => plan.eligible)}
-      />
+      </div>
+      {/* The eligibility modal needs to reinitialize on close hence the `isOpen &&` bit */}
+      {isOpen && (
+        <EligibilityModal
+          initialPlanIndex={initialPlanIndex}
+          onClose={closeModal}
+          eligibilityPlans={eligiblePlans}
+          isOpen
```

Comment:
Not sure if `isOpen` is useful since the display logic is no longer in `EligibilityModal`

Reply:
I will go ahead and remove it if everyone agrees